server/src/test/java/io/crate/integrationtests/ShardLimitsIT.java

```diff
@@ -65,6 +65,21 @@ partitioned by (p)
                 .hasHTTPError(HttpResponseStatus.BAD_REQUEST, 4000)
                 .hasMessageContaining(
                         "this action would add [4] total shards, but this cluster currently has [0]/[2] maximum shards open;");
+
+        // Bulk insert forcing creation of multiple partitions at once.
+        execute("set global \"cluster.max_shards_per_node\" = 4");
+        StringBuilder values = new StringBuilder();
+        for (int i = 1; i <= 20; i++) {
+            values.append("(" + i + "," + i + ")");
+            if (i < 20) {
+                values.append(",");
+            }
+        }
+        Asserts.assertSQLError(() -> execute("insert into tbl (x, p) values " + values))
+            .hasPGError(PGErrorStatus.INTERNAL_ERROR)
+            .hasHTTPError(HttpResponseStatus.BAD_REQUEST, 4000)
+            .hasMessageContaining(
+                "this action would add [80] total shards, but this cluster currently has [0]/[8] maximum shards open");
     }
```

Comment:
without the change, we end up with 80 shards in total (40 per node, instead of 4)